extensions/2.0/Vendor/GRIFFEL_bim_data/schema/GRIFFEL_bim_data.schema.json

```diff
@@ -0,0 +1,105 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "title": "GRIFFEL_bim_data Serializable Object",
+    "type": "object",
+    "description": "Contains same properties as embedded version plus properties to node map. Used only when data is saved in a separate file.",
+    "allOf": [ { "$ref": "glTFProperty.schema.json" } ],
+    "properties": {
+        "nodeProperties": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "description": "Node properties and type.",
+                "properties": {
+                    "node": {
+                        "allOf": [ { "$ref": "glTFid.schema.json" } ],
+                        "description": "Index of the node in glTF file."
+                    },
+                    "properties": {
+                        "type": "array",
+                        "item": {
+                            "allOf": [ { "$ref": "glTFid.schema.json" } ],
+                            "description": "Index of a property in the root level collection."
+                        },
+                        "description": "Collection of indices properties of the node.",
+                        "minItems": 1
+                    },
+                    "type":{
+                        "allOf": [ { "$ref": "glTFid.schema.json" } ],
+                        "description": "Index of the type of the node."
+                    }
+                },
+                "required": [ "properties" ]
```

Comment:
Given that elements of `nodeProperties` are supposed to link metadata to nodes, the set of required JSON properties should include `node`. On top of that, `type`, `properties`, or both should be defined for the mapping to have any effect (unless such data-less config has some special meaning).

The multiple-choice required properties could be expressed by putting the `required` lists inside `oneOf`.